lib/active_record/connection_adapters/mysql2_ghost_adapter.rb

```diff
@@ -55,14 +55,47 @@ def execute(sql, name = nil)
         end
       end
 
-      def add_index(table_name, column_name, options = {})
-        index_name, index_type, index_columns, index_options = add_index_options(table_name, column_name, options)
-        execute "ALTER TABLE #{quote_table_name(table_name)} ADD #{index_type} INDEX #{quote_column_name(index_name)} (#{index_columns})#{index_options}" # rubocop:disable Layout/LineLength
-      end
+      if Gem.loaded_specs['activerecord'].version >= Gem::Version.new('6.1')
```

Comment:
if there's a prettier way of achieving this, I'm all ears.  this was an easy solution that seems pretty simple to follow

Reply:
There are multiple ways of making this prettier:

```ruby
# in  lib/active_record/connection_adapters/mysql2_ghost_adapter/indexes_six.rb
module IndexesSix
  def add_index(....)
  end
end

# in  lib/active_record/connection_adapters/mysql2_ghost_adapter/indexes_five.rb
module IndexesFive
  def add_index(....)
  end
end

# in lib/active_record/connection_adapters/mysql2_ghost_adapter.rb
module ConnectionAdapters
    class Mysql2GhostAdapter < Mysql2Adapter
      if Gem.loaded_specs['activerecord'].version >= Gem::Version.new('6.1')
        include Mysql2GhostAdapter::IndexesSix
      else
        include Mysql2GhostAdapter::IndexesFive
      end
   end
end
```

Perhaps there's a more clever way but splitting into separate files will look a lot nicer.

Reply:
since the number of methods is relatively low, I somewhat prefer the existing way to including modules as its easier to navigate.  the second approach is nice, though, if we end up needing _another_ conditional.  I will make an issue for it as a pattern to consider if we also need 7.0 compatibility (somewhat likely)